src/main/io/serial_4way.c

```diff
@@ -464,36 +463,34 @@ void esc4wayProcess(serialPort_t *mspPort)
         // restart looking for new sequence from host
         do {
             CRC_in.word = 0;
-            timedOut = ReadByteCrc(&ESC, ESC_TIMEOUT_US);
-        } while ((ESC != cmd_Local_Escape) && !timedOut);
+            // No timeout as BLHeliSuite32 has this loops sitting indefinitely waiting for input
+            ReadByteCrc(&ESC, 0);
+        } while (ESC != cmd_Local_Escape);
 
         RX_LED_ON;
 
         Dummy.word = 0;
         O_PARAM = &Dummy.bytes[0];
         O_PARAM_LEN = 1;
 
+        timedOut = ReadByteCrc(&CMD, CMD_TIMEOUT_US) ||
+                   ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US) ||
+                   ReadByteCrc(&ioMem.D_FLASH_ADDR_L, ARG_TIMEOUT_US) ||
+                   ReadByteCrc(&I_PARAM_LEN, ARG_TIMEOUT_US);
```

Comment:
```suggestion
        timedOut = ReadByteCrc(&CMD, CMD_TIMEOUT_US) 
                   || ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US)
                   || ReadByteCrc(&ioMem.D_FLASH_ADDR_L, ARG_TIMEOUT_US) 
                   || ReadByteCrc(&I_PARAM_LEN, ARG_TIMEOUT_US);
```

Personally, I prefer putting operators on beginning of line - it is much easier to read when viewing code (you immediately see that it is 'or <something>' instead of backtracking to end of previous, usually long, line)

Reply:
i like the that idea.  grepping, i see both formats in source-code.

Reply:
https://stackoverflow.com/questions/28585760/recommendations-about-placement-of-logical-operators-in-multiline-condition-expr

In my opinion, readable code (operation/chaining is immediately obvious when quickly reading code) is preferable to nice alignment.

Or you can even make all lines uniform:
```suggestion
        timedOut = true
                   || ReadByteCrc(&CMD, CMD_TIMEOUT_US) 
                   || ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US)
                   || ReadByteCrc(&ioMem.D_FLASH_ADDR_L, ARG_TIMEOUT_US) 
                   || ReadByteCrc(&I_PARAM_LEN, ARG_TIMEOUT_US);
```

But it is just a minor detail ...

@nerdCopter : It is quite possible that lot of  'operator at beginning' are from my code. Most contributors may use other variant ..

Reply:
We should put it in our [coding style guide](http://betaflight.com/docs/development/CodingStyle)

https://www.gnu.org/prep/standards/html_node/Formatting.html

Reply:
Maybe I should force myself to change a well engrained preference!